pep-0695.rst

```diff
@@ -590,15 +605,10 @@ the defined type parameters.
 Accessing Type Parameters at Runtime
 ------------------------------------
 
-A new read-only attribute called ``__type_variables__`` is available on class,
-function, and type alias objects. This attribute is a tuple of the active
-type variables that are visible within the scope of that class, function,
-or type alias. This attribute is needed for runtime evaluation of stringified
-(forward referenced) type annotations that include references to type
-parameters. Functions like ``typing.get_type_hints`` can use this attribute
-to populate the ``locals`` dictionary with values for type parameters that
-are in scope when calling ``eval`` to evaluate the stringified expression.
-The tuple contains ``TypeVar`` instances.
+A new read-only attribute called ``__type_params__`` is available on generic class,
+function, and type alias objects. This attribute is a tuple of the
+type parameters that parameterize the class, function, or alias.
+The tuple contains ``TypeVar``, ``ParamSpec``, and ``TypeVarTuple`` instances.
```

Comment:
It is intentional that `__type_params__` only contains those typevars (etc.) introduced in this object's scope, right? As opposed to `__type_variables__` which included the visible ones from surrounding scopes. Because we no longer need those when we have PEP 649. Right?

Reply:
Yes, that's correct. My motivation is in https://github.com/erictraut/cpython/issues/10#issue-1679651222.